controllers/repository_controller.go

```diff
@@ -128,18 +130,20 @@ func (r *RepositoryReconciler) UpdateRepository(ctx context.Context, logger logr
 		return false, err
 	}
 
-	l = len(instanceDeepCopy.Status.URLHistory)
 	cond := corev1alpha1.Condition{
 		Status:             v1.ConditionTrue,
 		LastTransitionTime: metav1.Now(),
 		Reason:             corev1alpha1.ReasonCreating,
 		Message:            "prepare to launch watcher",
 		Type:               corev1alpha1.TypeReady,
 	}
-	instanceDeepCopy.Status.URLHistory = utils.AddString(instanceDeepCopy.Status.URLHistory, instanceDeepCopy.Spec.URL)
-	if len(instanceDeepCopy.Status.URLHistory) != l {
+
+	if swap, history := utils.AddOrSwapString(instanceDeepCopy.Status.URLHistory, instanceDeepCopy.Spec.URL); swap {
```

Comment:
should be `if !swap` ?

Reply:
The meaning of swap has been changed, swap is meant to express that the array has changed

Reply:
`swap` means you have replaced something like 

``` 
- a -> b
- b -> a
```

And seems you want to update status only when `swap` occured

Reply:
Yes, the code has been changed.
```go
if changed, history := utils.AddOrSwapString(); changed {}
```

Reply:
get it! both `swap` and`append`  will be treated as `changed`, then status got updated.